pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -155,18 +161,48 @@ func ConvertPolicyToJwtConfig(policy *authn.Policy) *jwtfilter.JwtAuthentication
 	return ret
 }
 
+// ConvertPolicyToAuthNFilterConfig returns authn filter config corresponding for the input policy.
+func ConvertPolicyToAuthNFilterConfig(policy *authn.Policy) *authn_filter.FilterConfig {
+	if policy == nil {
+		return &authn_filter.FilterConfig{Policy: &authn.Policy{}}
+	}
+	filterConfig := &authn_filter.FilterConfig{
+		Policy: &authn.Policy{
```

Comment:
why don't we just do:

     Policy:  policy

Reply:
I need to delete the targets part of the policy, and much as I know about go, policy is a pointer so it actually modify the original one (which break the pilot test, when we reuse policy in for different tests).

Anyway, I make a change here to clone the message and delete the unwanted part.